microservices-connector/cmd/router/main.go

```diff
@@ -198,6 +199,32 @@ func executeStep(
 	return callService(step, serviceURL, input, headers)
 }
 
+func mergeRequests(respReq []byte, initReqData map[string]interface{}) []byte {
+	var respReqData map[string]interface{}
+
+	if _, exists := initReqData[Parameters]; exists {
+		if err := json.Unmarshal(respReq, &respReqData); err != nil {
+			log.Error(err, "Error unmarshaling respReqData:")
+			return nil
+		}
+		// Merge init request into respReq
+		for key, value := range initReqData[Parameters].(map[string]interface{}) {
+			/*if _, exists := respReqData[key]; !exists {
```

Comment:
remove this if not needed

Reply:
Hi @KfreeZ, just let's keep it there, because let's see if they need to keep the response instead overwriting by initial request.